src/Admin/PageAdmin.php

```diff
@@ -196,7 +198,9 @@ protected function configureDatagridFilters(DatagridMapper $filter): void
             ->add('parent')
             ->add('edited')
             ->add('hybrid', CallbackFilter::class, [
-                'callback' => static function ($queryBuilder, $alias, $field, $data): void {
+                'callback' => static function (ProxyQueryInterface $queryBuilder, string $alias, string $field, array $data): void {
+                    \assert($queryBuilder instanceof ProxyQuery);
```

Comment:
Can't we use ProxyQuery in the typehint of the callback then ?

Reply:
I followed docs of DoctrineORMAdminBundle: https://docs.sonata-project.org/projects/SonataDoctrineORMAdminBundle/en/4.x/reference/filter_field_definition/#callback.

Maybe those docs need an update?

What I don't like about that ProxyQueryInterface is that is specific for the DoctrineORMAdminBundle but does not represent all the method you can call for the real object... IMO is something that could be fixed for 5.0.

For this PR, do you prefer to change typehint?

Reply:
> What I don't like about that ProxyQueryInterface is that is specific for the DoctrineORMAdminBundle but does not represent all the method you can call for the real object... IMO is something that could be fixed for 5.0.

I personally dont like the `__call`, all the method you can call on the real object are just the same than `->getQueryBuilder->yourMethod()`. But I didn't removed it because I assume a lot of people are using it.

I think you wont need an assert if you call getQueryBuilder first

Reply:
Fixed, probably we could deprecate the __call in favour of getQueryBuilder.